internal/services/embedded_wallet_service.go

```diff
@@ -173,3 +181,62 @@ func (e *EmbeddedWalletService) GetWalletByCredentialID(ctx context.Context, cre
 		return embeddedWallet, nil
 	})
 }
+
+func (e *EmbeddedWalletService) SponsorTransaction(ctx context.Context, account, operationXDR string) (string, error) {
+	if account == "" {
+		return "", ErrMissingAccount
+	}
+	if operationXDR == "" {
+		return "", ErrMissingOperationXDR
+	}
+
+	return db.RunInTransactionWithResult(ctx, e.sdpModels.DBConnectionPool, nil, func(sdpTx db.DBTransaction) (string, error) {
+		insert := data.SponsoredTransactionInsert{
+			ID:           uuid.New().String(),
+			Account:      account,
+			OperationXDR: operationXDR,
+			Status:       data.PendingSponsoredTransactionStatus,
+		}
+		sponsoredTx, err := e.sdpModels.SponsoredTransactions.Insert(ctx, sdpTx, insert)
+		if err != nil {
+			return "", fmt.Errorf("creating sponsored transaction: %w", err)
+		}
+
+		currentTenant, err := sdpcontext.GetTenantFromContext(ctx)
+		if err != nil {
+			return "", fmt.Errorf("getting tenant from context: %w", err)
+		}
+
+		tssTransaction := &store.Transaction{
+			ExternalID:      sponsoredTx.ID,
+			TransactionType: store.TransactionTypeSponsored,
+			TenantID:        currentTenant.ID,
+			Sponsored: store.Sponsored{
+				SponsoredAccount:        account,
+				SponsoredTransactionXDR: operationXDR,
+			},
+		}
+
+		_, err = e.tssModel.BulkInsert(ctx, e.tssModel.DBConnectionPool, []store.Transaction{*tssTransaction})
```

Comment:
```suggestion
		_, err = e.tssModel.BulkInsert(ctx, sdpTx, []store.Transaction{*tssTransaction})
```
Shouldn't we use the sdpTx to keep this within the transaction boundary?